kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -135,3 +93,56 @@ func enableComputeAPI(project string) error {
 
 	return nil
 }
+
+// getProjectFromBoskos creates a boskos client, acquires a gcp project
+// and starts a heartbeat goroutine to keep the project reserved
+func (d *deployer) getProjectFromBoskos() error {
+	boskos, err := client.NewClient(
+		os.Getenv("JOB_NAME")+"-kubetest2",
+		d.BoskosLocation,
+		"",
+		"",
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create boskos client: %s", err)
+	}
+
+	resourceType := "gce-project"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	boskosProject, err := boskos.AcquireWait(ctx, resourceType, "free", "busy")
+	if err != nil {
+		return fmt.Errorf("failed to get a %s from boskos: %s", resourceType, err)
+	}
+	if boskosProject == nil {
+		return fmt.Errorf("boskos had no %s available", resourceType)
+	}
+
+	go func(c *client.Client, resource *boskosCommon.Resource) {
```

Comment:
Will this goroutine ever get cleaned up? May want to use a Done channel to allow explicit closing and return a close func. Even better, issue an explicit release to boskos if closed, say at teardown

Reply:
It's definitely not the cleanest way to do this. Updated with an explicit close after the project is released.